cloud/aws/actuators/machine/actuator.go

```diff
@@ -498,6 +500,14 @@ func (a *Actuator) updateStatus(machine *clusterv1.Machine, instance *ec2.Instan
 	} else {
 		mLog.Debug("status unchanged")
 	}
+
+	// If machine state is still pending, we will return an error to keep the controllers
+	// attempting to update status until it hits a more permanent state. This will ensure
+	// we get a public IP populated more quickly.
+	if awsStatus.InstanceState != nil && *awsStatus.InstanceState == ec2.InstanceStateNamePending {
```

Comment:
Your pr got merged upstream :) https://github.com/kubernetes-sigs/cluster-api/commit/35d26d068f82830253eb24eb659f4babb22380cb
Would this work for the case when a machine is being rebooted? or should we may be send another ec2 state to requeue?

Reply:
Awesome!

Hmm I wonder, does the cluster API monitor node state? Once a node is registered if we watched nodes and then resynced their associated machines, it would work. I'm not sure if cluster-api does that today.